hip_data_tools/aws/redshift.py

```diff
@@ -0,0 +1,448 @@
+import re
+
+import boto3
+import psycopg2
+
+DTYPE_REDSHIFT_TO_CSV_ATHENA = {
+    # Numeric
+    "INTEGER": "BIGINT",
+    "BIGINT": "BIGINT",
+    "DOUBLE": "DOUBLE",
+    "SMALLINT": "BIGINT",
+    "REAL": "BIGINT",
+    "NUMERIC": "DOUBLE",
+
+    # Date & time
+    "TIMESTAMP": "STRING",
+
+    # Strings
+    "BOOLEAN": "STRING",
+    "DATE": "STRING",
+    "CHARACTER": "STRING"
+}
+
+DTYPE_REDSHIFT_TO_PARQUET_ATHENA = {
+    # Numeric
+    "INTEGER": "INTEGER",
+    "BIGINT": "BIGINT",
+    "DOUBLE": "DOUBLE",
+    "SMALLINT": "SMALLINT",
+    "REAL": "BIGINT",
+    "NUMERIC": "DECIMAL",
+    "BOOLEAN": "BOOLEAN",
+    "CHARACTER": "STRING",
+
+    # Date & time
+    "TIMESTAMP": "TIMESTAMP",
+    "DATE": "DATE"
+}
+
+
+class RedshiftUtil:
+    """
+    Utility to read, write and interact with redshift.
+    The environments are isolated with schemas.
+    there are convenience methods for most types of transformations and interactions.
+    """
+
+    def __init__(self, host, port, schema, user, password):
```

Comment:
consider making into a redshift connection class and pass in the connection object instead of handling the username passwords in this class.